pkg/core/mempool/mem_pool.go

```diff
@@ -228,12 +229,13 @@ func (mp *Pool) Remove(hash util.Uint256) {
 // drop part of the mempool that is now invalid after the block acceptance.
 func (mp *Pool) RemoveStale(isOK func(*transaction.Transaction) bool, feer Feer) {
 	mp.lock.Lock()
+	policyChanged := mp.loadPolicy(feer)
 	// We can reuse already allocated slice
 	// because items are iterated one-by-one in increasing order.
 	newVerifiedTxes := mp.verifiedTxes[:0]
 	mp.fees = make(map[util.Uint160]utilityBalanceAndFees) // it'd be nice to reuse existing map, but we can't easily clear it
 	for _, itm := range mp.verifiedTxes {
-		if isOK(itm.txn) && mp.tryAddSendersFee(itm.txn, feer) {
+		if isOK(itm.txn) && mp.checkPolicy(itm.txn, policyChanged) && mp.tryAddSendersFee(itm.txn, feer) {
```

Comment:
It's not correct, the policy might have changed, but we may still have enough gas in fees to pay for transaction.

Reply:
`checkPolicy` simply checks, whether transaction's `feePerByte` (which is `NetworkFee / size`) is greater than or equal to policy's `feePerByte`. It's true that after policy have changed we may still be able to pay for it, but we can't modify the transaction (i.e. its `NetworkFee`), so it will have the same `feePerByte` and there's a chance that this value won't fit new `feePerByte` from Policy. Not quite understand why it's not correct.

Reply:
And it is, it's my view of this code that was incorrect, fee per byte value computed as fee/size can be compared this way.